staging/src/k8s.io/client-go/tools/cache/mutation_detector.go

```diff
@@ -113,8 +113,12 @@ func (d *defaultCacheMutationDetector) CompareObjects() {
 
 	altered := false
 	for i, obj := range d.cachedObjs {
-		if !reflect.DeepEqual(obj.cached, obj.copied) {
-			fmt.Printf("CACHE %s[%d] ALTERED!\n%v\n", d.name, i, diff.ObjectGoPrintSideBySide(obj.cached, obj.copied))
+		// make sure that we have a stable view of the cached object
+		// sometimes code will modify an object and then reset it back to the original state
+		// this can result in reflect.DeepEqual returning false while cmp.Diff prints an empty diff
+		cached := obj.cached.DeepCopyObject()
+		if !reflect.DeepEqual(cached, obj.copied) {
```

Comment:
this is even too weak as an equality. We want pointer-equality, not just value equality.

Reply:
We compare against a deep copy. Pointers won't be equal. But there is a chance that we miss mutations if they change pointers to extensionally equal data. But not much we can do, and that restriction was here already before the PR.